app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.js

```diff
@@ -81,20 +82,23 @@ export function validateAddEthereumChainParams(params, end) {
     });
   }
 
-  const {
-    chainId,
-    chainName,
-    blockExplorerUrls,
-    nativeCurrency,
-    rpcUrls,
-    ...otherParams
-  } = params;
-
-  if (Object.keys(otherParams).length > 0) {
+  const otherKeys = Object.keys(
+    omit(params, [
+      'chainId',
+      'chainName',
+      'blockExplorerUrls',
+      'nativeCurrency',
+      'rpcUrls',
+      'iconUrls',
+    ]),
+  );
```

Comment:
thoughts on keeping `...otherParams` and omitting against that instead that way we can avoid repeating the keys in the omit and destructing assignment?

```
  const {
    chainId,
    chainName,
    blockExplorerUrls,
    nativeCurrency,
    rpcUrls,
    ...otherParams
  } = params;
  
  const otherKeys = Object.keys().filter(v => !['iconUrls'].includes(v));

  if (Object.keys(otherKeys).length > 0) {

```

feel free to disregard

Reply:
Yeah I considered this too. I'm fine with either way.

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/24920/commits/1a50c849c354ed56706a3ce7c1d7eadc72d62d1f